oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -262,6 +275,7 @@ public static Builder withRegistrationId(String registrationId) {
 		private AuthenticationMethod userInfoAuthenticationMethod = AuthenticationMethod.HEADER;
 		private String userNameAttributeName;
 		private String jwkSetUri;
+		private Map<String, Object> configurationMetadata;
```

Comment:
I think it would be good to make this by default empty so user's don't need to worry about NPE